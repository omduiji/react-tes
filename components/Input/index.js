import React from 'react'
import styles from './styles.module.scss'

const Input = ({ children, label, value, onChange = () => {} }) => {
    if (!label) throw Error('Label is required')

    const id = label.toLowerCase().replace(/ /g, '_')
    return (
        <fieldset className={styles.fieldSet}>
        {label && <label htmlFor={id} className={styles.label}>{label}</label>}
        <input
            type="text"
            className={styles.input}
            onChange={e => onChange(e.target.value)}
            value={value}
            id={id}
        />
        </fieldset>
    )
    }

export default Input
