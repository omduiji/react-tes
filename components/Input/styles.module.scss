.fieldSet {
    border: none;
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 250px;
}

.label {
    font-size: 1.2rem;
    margin: 0.75em 0 0.25em 0;
    text-align: left;
}

.input {
    margin: 0.75em 0;
    padding: 0 7px;
    width: 100%;
}

.label + .input {
    margin-top: 0;
}
