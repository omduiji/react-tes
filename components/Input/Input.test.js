import React from 'react'
import { render } from '@testing-library/react'
import Input from '.'

test('should render Input', () => {
  const { asFragment } = render(<Input label="test label" />)

  expect(asFragment()).toMatchSnapshot()
})
