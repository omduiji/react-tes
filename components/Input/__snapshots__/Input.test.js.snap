// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input 1`] = `
<DocumentFragment>
  <fieldset
    class="fieldSet"
  >
    <label
      class="label"
      for="test_label"
    >
      test label
    </label>
    <input
      class="input"
      id="test_label"
      type="text"
      value=""
    />
  </fieldset>
</DocumentFragment>
`;
